### ISSUE TITLE
Restart Required After Install

### ISSUE THREAD

**Issue by jazeren1** [TWCable/grabbit#11]
After installing the Grabbit project onto CQ instances by running the gradle command:
"gradlew install"
The CQ instances that were installed to leave the Spring Batch Infrastructure bundle in the "Installed" state. Because Spring Batch is required to run Grabbit, Grabbit is unusable at this point.
The current work around is simply to restart the CQ instances and Spring Batch starts up correctly. However, this is not a long term solution and needs to be addressed.

**Comment by jdigger** [comment]
Error in log:
```
27.03.2015 13:45:05.521 ERROR [0:0:0:0:0:0:0:1 [1427463905517] POST /system/console/bundles/354 HTTP/1.1] cqse-httpservice %bundles.pluginTitle: Cannot start (org.osgi.framework.BundleException: Uses constraint violation. Unable to resolve bundle revision org.springframework.batch.infrastructure [354.0] because it exports package 'org.springframework.batch.repeat' and is also exposed to it from bundle revision org.springframework.batch.infrastructure [327.0] via the following dependency chain:

org.springframework.batch.infrastructure [354.0]
import: (&(osgi.wiring.package=org.springframework.batch.core)(version>=2.0.0)(!(version>=3.0.0)))
|
export: osgi.wiring.package=org.springframework.batch.core; uses:=org.springframework.batch.repeat
org.springframework.batch.core [353.0]
import: (&(osgi.wiring.package=org.springframework.batch.repeat)(version>=2.2.0)(!(version>=3.0.0)))
|
export: osgi.wiring.package=org.springframework.batch.repeat
org.springframework.batch.infrastructure [327.0]) org.osgi.framework.BundleException: Uses constraint violation. Unable to resolve bundle revision org.springframework.batch.infrastructure [354.0] because it exports package 'org.springframework.batch.repeat' and is also exposed to it from bundle revision org.springframework.batch.infrastructure [327.0] via the following dependency chain:

org.springframework.batch.infrastructure [354.0]
import: (&(osgi.wiring.package=org.springframework.batch.core)(version>=2.0.0)(!(version>=3.0.0)))
|
export: osgi.wiring.package=org.springframework.batch.core; uses:=org.springframework.batch.repeat
org.springframework.batch.core [353.0]
import: (&(osgi.wiring.package=org.springframework.batch.repeat)(version>=2.2.0)(!(version>=3.0.0)))
|
export: osgi.wiring.package=org.springframework.batch.repeat
org.springframework.batch.infrastructure [327.0]
at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:3968)
at org.apache.felix.framework.Felix.startBundle(Felix.java:2025)
at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:955)
at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:942)
at org.apache.felix.webconsole.internal.core.BundlesServlet.doPost(BundlesServlet.java:364)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:727)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:820)
at org.apache.felix.webconsole.internal.servlet.OsgiManager.service(OsgiManager.java:526)
at org.apache.felix.webconsole.internal.servlet.OsgiManager.service(OsgiManager.java:450)
at org.apache.felix.http.base.internal.handler.ServletHandler.doHandle(ServletHandler.java:96)
at org.apache.felix.http.base.internal.handler.ServletHandler.handle(ServletHandler.java:79)
at org.apache.felix.http.base.internal.dispatch.ServletPipeline.handle(ServletPipeline.java:42)
at org.apache.felix.http.base.internal.dispatch.InvocationFilterChain.doFilter(InvocationFilterChain.java:49)
at org.apache.felix.http.base.internal.dispatch.HttpFilterChain.doFilter(HttpFilterChain.java:33)
at org.apache.sling.i18n.impl.I18NFilter.doFilter(I18NFilter.java:127)
at org.apache.felix.http.base.internal.handler.FilterHandler.doHandle(FilterHandler.java:88)
at org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:76)
at org.apache.felix.http.base.internal.dispatch.InvocationFilterChain.doFilter(InvocationFilterChain.java:47)
at org.apache.felix.http.base.internal.dispatch.HttpFilterChain.doFilter(HttpFilterChain.java:33)
at org.apache.felix.http.sslfilter.internal.SslFilter.doFilter(SslFilter.java:55)
at org.apache.felix.http.base.internal.handler.FilterHandler.doHandle(FilterHandler.java:88)
at org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:76)
at org.apache.felix.http.base.internal.dispatch.InvocationFilterChain.doFilter(InvocationFilterChain.java:47)
at org.apache.felix.http.base.internal.dispatch.HttpFilterChain.doFilter(HttpFilterChain.java:33)
at com.adobe.granite.license.impl.LicenseCheckFilter.doFilter(LicenseCheckFilter.java:179)
at org.apache.felix.http.base.internal.handler.FilterHandler.doHandle(FilterHandler.java:88)
at org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:76)
at org.apache.felix.http.base.internal.dispatch.InvocationFilterChain.doFilter(InvocationFilterChain.java:47)
at org.apache.felix.http.base.internal.dispatch.HttpFilterChain.doFilter(HttpFilterChain.java:33)
at org.apache.sling.security.impl.ReferrerFilter.doFilter(ReferrerFilter.java:263)
at org.apache.felix.http.base.internal.handler.FilterHandler.doHandle(FilterHandler.java:88)
at org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:76)
at org.apache.felix.http.base.internal.dispatch.InvocationFilterChain.doFilter(InvocationFilterChain.java:47)
at org.apache.felix.http.base.internal.dispatch.HttpFilterChain.doFilter(HttpFilterChain.java:33)
at org.apache.sling.engine.impl.log.RequestLoggerFilter.doFilter(RequestLoggerFilter.java:75)
at org.apache.felix.http.base.internal.handler.FilterHandler.doHandle(FilterHandler.java:88)
at org.apache.felix.http.base.internal.handler.FilterHandler.handle(FilterHandler.java:76)
at org.apache.felix.http.base.internal.dispatch.InvocationFilterChain.doFilter(InvocationFilterChain.java:47)
at org.apache.felix.http.base.internal.dispatch.HttpFilterChain.doFilter(HttpFilterChain.java:33)
at org.apache.felix.http.base.internal.dispatch.FilterPipeline.dispatch(FilterPipeline.java:48)
at org.apache.felix.http.base.internal.dispatch.Dispatcher.dispatch(Dispatcher.java:39)
at org.apache.felix.http.base.internal.DispatcherServlet.service(DispatcherServlet.java:67)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:820)
at com.day.j2ee.servletengine.ServletRuntimeEnvironment.service(ServletRuntimeEnvironment.java:250)
at com.day.j2ee.servletengine.RequestDispatcherImpl.doFilter(RequestDispatcherImpl.java:321)
at com.day.j2ee.servletengine.RequestDispatcherImpl.service(RequestDispatcherImpl.java:340)
at com.day.j2ee.servletengine.RequestDispatcherImpl.service(RequestDispatcherImpl.java:383)
at com.day.j2ee.servletengine.ServletHandlerImpl.process(ServletHandlerImpl.java:335)
at com.day.j2ee.servletengine.HttpListener$Worker.run(HttpListener.java:644)
at java.lang.Thread.run(Thread.java:745)
```

**Comment by jdigger** [comment]
A DayCare (support) ticket was filed with Adobe on April 6. Still waiting to hear back.

**Comment by jdigger** [comment]
The Felix bug has been identified and has been fixed in Felix 5.2: https://issues.apache.org/jira/browse/FELIX-4867

The work-around until then is to "refresh all bundles" or restart Felix.
